include/cudaq/Optimizer/Dialect/Quake/QuakeOps.td

```diff
@@ -754,6 +754,23 @@ class TwoTargetOp<string mnemonic, list<Trait> traits = []> :
 // Quantum operators (gates)
 //===----------------------------------------------------------------------===//
 
+def ExpPauliOp : QuakeOp<"exp_pauli", []> {
+  let summary = "General Pauli tensor product rotation";
+  let description = [{
+    This operation affects a general Pauli tensor product rotation on 
+    the input qubits. The number of Pauli characters in the input Pauli word 
+    string must equal the number of qubits in the veq. Mathematically, this operation 
+    applies exp(-i theta P) where P is a general Pauli tensor product. 
```

Comment:
From this description, I'm thinking `std::string` is maybe a bit heavyweight for this. We probably want the programming interface to take an actual string literal and add the template constraints to make sure it has the correct size per the number of arguments, etc.

Then at the quake level, we can just use a type of `!cc.array<i8 x `_n_`>` for a C-string literal.

Reply:
I think the counter-argument (and there is an example in the tests for this) is the case where the user wants to pass the string as a kernel argument.

Reply:
The counterargument is `std::string` is not a literal nor the type of a literal. It's a C++ class (actually a template specialization of a parametric polymorphic class) with allocators, data members, a number of methods for doing things with strings, iterators, hashing, etc. Supporting all that "extra stuff" doesn't seem necessary. Yet the implication of adding a `std::string` type to quake is that quake will possibly support those features.

Reply:
I see your point - I will update to remove `cc.string` and have `cc.string_literal` return a `cc.array<i8*N>` and input `std::string` kernel arguments to `cc.ptr<i8>`.